client/extensions/woocommerce/app/settings/payments/stripe/payment-method-stripe-key-based-dialog.js

```diff
@@ -0,0 +1,240 @@
+					value: method.settings.secret_key.value,
+				},
+				{
+					id: 'publishable_key',
+					label: translate( 'Live Publishable Key' ),
+					placeholder: translate( 'Enter your publishable key from your Stripe.com account' ),
+					value: method.settings.publishable_key.value,
+				}
+			];
+		} else {
+			keys = [
+				{
+					id: 'test_secret_key',
+					label: translate( 'Test Secret Key' ),
+					placeholder: translate( 'Enter your secret key from your Stripe.com account' ),
+					value: method.settings.test_secret_key.value,
+				},
+				{
+					id: 'test_publishable_key',
+					label: translate( 'Test Publishable Key' ),
+					placeholder: translate( 'Enter your publishable key from your Stripe.com account' ),
+					value: method.settings.test_publishable_key.value,
+				}
+			];
+		}
+
+		return (
+			<div>
+				<FormFieldset className="stripe__method-edit-field-container">
+					<TestLiveToggle
+						isTestMode={ 'yes' === method.settings.testmode.value }
+						onSelectLive={ this.onSelectLive }
+						onSelectTest={ this.onSelectTest }
+					/>
+				</FormFieldset>
+				<APIKeysView
+					highlightEmptyRequiredFields={ this.state.highlightEmptyRequiredFields }
+					keys={ keys }
+					onEdit={ this.props.onEditField }
+				/>
+			</div>
+		);
+	}
+
+	onSelectAuthOnly = () => {
+		this.props.onEditField( { target: { name: 'capture', value: 'no' } } );
+	}
+
+	onSelectCapture = () => {
+		this.props.onEditField( { target: { name: 'capture', value: 'yes' } } );
+	}
+
+	getStatementDescriptorPlaceholder = () => {
+		const { site, translate } = this.props;
+		const domain = site.domain.substr( 0, 22 ).trim().toUpperCase();
+		return translate( 'e.g. %(domain)s', { args: { domain } } );
+	}
+
+	renderMoreSettings = () => {
+		const { method, translate } = this.props;
+
+		return (
+			<div>
+				<AuthCaptureToggle
+					isAuthOnlyMode={ 'no' === method.settings.capture.value }
+					onSelectAuthOnly={ this.onSelectAuthOnly }
+					onSelectCapture={ this.onSelectCapture }
+				/>
+				<FormFieldset>
+					<FormLabel>
+						{ translate( 'Descriptor' ) }
+					</FormLabel>
+					<FormTextInput
+						name="statement_descriptor"
+						onChange={ this.props.onEditField }
+						value={ method.settings.statement_descriptor.value }
+						placeholder={ this.getStatementDescriptorPlaceholder() } />
+					<FormSettingExplanation>
+						{ translate( 'Appears on your customer\'s credit card statement. 22 characters maximum' ) }
+					</FormSettingExplanation>
+				</FormFieldset>
+				<FormFieldset className="stripe__method-edit-field-container">
+					<FormLabel>
+						{ translate( 'Use Apple Pay' ) }
+					</FormLabel>
+					<PaymentMethodEditFormToggle
+						checked={ method.settings.apple_pay.value === 'yes' ? true : false }
+						name="apple_pay"
+						onChange={ this.props.onEditField } />
+					<span>
+						{ translate(
+							'By using Apple Pay you agree to Stripe and ' +
```

Comment:
Do we need a link to the TOS here eventually?

Reply:
Eventually yes - there is a separate issue for that